skunkworks_crow/src/main/java/org/odk/share/bluetooth/BluetoothReceiver.java

```diff
@@ -0,0 +1,60 @@
+package org.odk.share.bluetooth;
+
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+
+import timber.log.Timber;
+
+/**
+ * @author huangyz0918 (huangyz0918@gmail.com)
+ * @since 04/06/2019
+ */
+public class BluetoothReceiver extends BroadcastReceiver {
+
+    private static final String TAG = BluetoothReceiver.class.getSimpleName();
+    private final BluetoothReceiverListener bluetoothReceiverListener;
+
+    public BluetoothReceiver(Context context, BluetoothReceiverListener bluetoothReceiverListener) {
+        this.bluetoothReceiverListener = bluetoothReceiverListener;
+        IntentFilter filter = new IntentFilter();
+        filter.addAction(BluetoothAdapter.ACTION_STATE_CHANGED); //Bluetooth switch status.
+        filter.addAction(BluetoothDevice.ACTION_FOUND); //Bluetooth discovers new devices (unpaired devices).
+        context.registerReceiver(this, filter);
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String action = intent.getAction();
+        if (action == null) {
+            return;
+        }
+
+        Timber.d(TAG, action);
+        BluetoothDevice dev = intent.getParcelableExtra(BluetoothDevice.EXTRA_DEVICE);
+        if (dev != null) {
```

Comment:
And what if dev is null, on `deviceFound `should not be called.

Reply:
Fixed.